Practice Areas/Engineering/Backend/Sentry Tagging Standards.md

```diff
@@ -0,0 +1,63 @@
+
+# VSP Sentry Tagging Standards
+
+## Background
+
+Sentry allows for adding [additional context](https://docs.sentry.io/enriching-error-data/context/?platform=ruby) to captured error events. 
+
+In the context of the VSP, we aim to standardize our use of these features for consistentency and reliability in searching/filtering Sentry events.
+
+**Definitions**
+*Tags* are key/value pairs which generate breakdowns charts and [search filters](http://sentry.vfs.va.gov/vets-gov/platform-api-production/?query=is%3Aunresolved+backend_service%3Aevss). You can see all tags for the given Sentry project [here](http://sentry.vfs.va.gov/settings/vets-gov/projects/platform-api-production/tags/). Tags will also be used in the Sentry alert sent to Slack.
+
+*Extra* context is unstructured and difficult to use as a search/filter metric. Use sparingly.
+
+
+
+### What you get for free:
+- every controller action that inherits from ApplicationController is tagged with:
+    ```ruby
+     def user_context
+        {
+          uuid: current_user&.uuid,
+          authn_context: current_user&.authn_context,
+          loa: current_user&.loa,
+          mhv_icn: current_user&.mhv_icn
+        }
+      end
+
+      def tags_context
+
+        {
+          controller_name: controller_name,
+          sign_in_method: sign_in_method_for_tag
+        }
+      end
+
+      def extra_context
+        {
+          request_uuid: request.uuid
+        }
+      end
+    ```
+
+`job`
+- Sidekiq job is automatically tagged in `lib/sidekiq/error_tag.rb` with the job class name
+
+### Manually added tag guidelines:
+
+The following tags are available and encouraged for use in service transactions. 
+
+Generally it is advisable to provide warrants if adding a new tag not seen here in a given PR.
```

Comment:
I don't quite understand this sentence - consider rewording? Specifically, I don't know what "warrants" mean in this context.

Reply:
Good call. This is a bit more straight forward:

```suggestion
Tags can be arbitrarily added, so we should be careful to use a common pattern. If you wish to use a tag that you don't see listed here, please provide sufficient reasoning for its use in your pull request.
```